app/components/post_list/combined_user_activity/combined_user_activity.tsx

```diff
@@ -0,0 +1,249 @@
+import SystemAvatar from '@components/post_list/system_avatar';
+import SystemHeader from '@components/post_list/system_header';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {Posts} from '@mm-redux/constants';
+import {emptyFunction} from '@utils/general';
+import {getMarkdownTextStyles} from '@utils/markdown';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type {Post} from '@mm-redux/types/posts';
+import type {Theme} from '@mm-redux/types/preferences';
+
+import LastUsers from './last_users';
+import {postTypeMessages} from './messages';
+
+type Props = {
+    canDelete: boolean;
+    currentUserId?: string;
+    currentUsername?: string;
+    intl: typeof intlShape;
+    post: Post;
+    showJoinLeave: boolean;
+    testID?: string;
+    theme: Theme;
+    userProfiles: Record<string, string>;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        baseText: {
+            color: theme.centerChannelColor,
+            opacity: 0.6,
+        },
+        body: {
+            flex: 1,
+            paddingBottom: 2,
+            paddingTop: 2,
+        },
+        container: {
+            flexDirection: 'row',
+        },
+        content: {
+            flex: 1,
+            flexDirection: 'column',
+            marginRight: 12,
+        },
+        displayName: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            fontWeight: '600',
+            flexGrow: 1,
+            paddingVertical: 2,
+        },
+        displayNameContainer: {
+            maxWidth: '60%',
+            marginRight: 5,
+            marginBottom: 3,
+        },
+        header: {
+            flex: 1,
+            flexDirection: 'row',
+            marginTop: 10,
+        },
+        profilePictureContainer: {
+            marginBottom: 5,
+            marginLeft: 12,
+            marginRight: 13,
+            marginTop: 10,
+        },
+        time: {
+            color: theme.centerChannelColor,
+            fontSize: 12,
+            marginTop: 5,
+            opacity: 0.5,
+            flex: 1,
+        },
+    };
+});
+
+const CombinedUserActivity = ({
+    canDelete, currentUserId, currentUsername, intl, post,
+    showJoinLeave, testID, theme, userProfiles,
+}: Props) => {
+    const itemTestID = `${testID}.${post.id}`;
+    const textStyles = getMarkdownTextStyles(theme);
+    const {allUsernames, messageData} = post.props.user_activity;
+    const styles = getStyleSheet(theme);
+    const content = [];
+    const removedUserIds: string[] = [];
+
+    const getUsernames = (userIds: string[]) => {
+        const someone = intl.formatMessage({id: 'channel_loader.someone', defaultMessage: 'Someone'});
+        const usernames = userIds.reduce((acc: string[], id: string) => {
+            if (id !== currentUserId && id !== currentUsername) {
+                const name = userProfiles[id];
+                if (name) {
```

Comment:
```suggestion
                acc.push(name ? `@${name}` : someone)
```

We can reduce the if-else clause